src/validation/joi/joi.shared.schemas.ts

```diff
@@ -13,6 +13,7 @@ export const booleanDefaultToFalseSchema = Joi.boolean().default(false) as Boole
 export const stringSchema = Joi.string()
 export const numberSchema = Joi.number()
 export const integerSchema = Joi.number().integer()
+export const percentageSchema = Joi.number().min(0).max(100)
```

Comment:
I'd prefer it to be integers (you know that I like integers, right?😀) 

```suggestion
export const percentageSchema = Joi.integer().min(0).max(100)
```

Reply:
Hmm are you sure? I think `12.5` should be allowed 🤓

Reply:
`ouraSleepData` has number 😄  `efficiency: numberSchema.min(0).max(100),` But I see that is commited by Kristofer 😂  #teamnumber

Reply:
I really dislike Floats, in places where Integers do just fine. Floats cause the whole class of issues (like `10/3*3`), so, things like BigDecimal, BigInt and storing Money as cents are invented.

If you want - you can do separate `integerPercentageSchema` / `floatPercentageSchema` (or just `percentageSchema`). But I'd use `integerPercentageSchema` as much as possible, to avoid Float issues.

Reply:
Okay, I guess in cases where we need to handle more precise percentages we can use a custom schema :)

Reply:
Yes.

I also found one more argument since yesterday. In cases where we need "float percentages", we can as well use "fractions of 1" instead of percentages. Like `Math.random` gives a number from 0 to 1. So, 0.125 will mean the same as 12.5%. It's quite common in Math/science.

Reply:
Yeah as I said in our meeting on monday I prefer 0-1 as percentage 🤓